pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +190,75 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def encoder(in_channels: int = 3,) -> enc.SequentialEncoder:
+    r"""Encoder part of the :class:`Transformer` from :cite:`SKL+2018`.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+
+    """
+    modules = [
+        nn.ReflectionPad2d(15),
+        ConvBlock(in_channels=in_channels, out_channels=32, kernel_size=3, stride=1),
+    ]
+    modules.extend(
+        channel_progression(
+            lambda in_channels, out_channels: ConvBlock(
+                in_channels, out_channels, kernel_size=3, stride=2
+            ),
+            channels=(32, 32, 64, 128, 256),
+        )
+    )
+    return enc.SequentialEncoder(modules)
+
+
+class ValueRangeDelimiter(pystiche.Module):
+    r"""Maps the values to the interval (-1.0, 1.0)."""
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        return torch.tanh(input / 2)
+
+    def _properties(self) -> Dict[str, Any]:
+        dct = super()._properties()
+        dct["output_range"] = (-1.0, 1.0)
+        return dct
+
+
+def decoder(
+    out_channels: int = 3, num_residual_blocks: int = 9,
+) -> pystiche.SequentialModule:
+    r"""Decoder part of the :class:`Transformer` from :cite:`SKL+2018`."""
+    residual_blocks = [residual_block(256) for _ in range(num_residual_blocks)]
+    upsample_conv_blocks = channel_progression(
+        lambda in_channels, out_channels: UpsampleConvBlock(
+            in_channels, out_channels, kernel_size=3
+        ),
+        channels=(256, 256, 128, 64, 32),
+    )
+    return pystiche.SequentialModule(
+        *residual_blocks,
+        *upsample_conv_blocks,
+        nn.ReflectionPad2d(3),
+        conv(
+            in_channels=32,
+            out_channels=out_channels,
+            kernel_size=7,
+            stride=1,
+            padding="valid",
+        ),
+        ValueRangeDelimiter(),
+    )
+
+
+class Transformer(nn.Module):
```

Comment:
Could you add a function `transformer` that returns an instance of this in compliance with the others parts?